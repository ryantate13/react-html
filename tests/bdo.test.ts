import {bdo} from '../src';

describe('bdo', () => {
    it('renders with children', () => {
        expect(bdo({}, 'TEST')).toMatchSnapshot();
    });
    
    it('renders arrays of children', () => {
        expect(bdo({}, Array(10).fill('TEST'))).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(bdo({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(bdo()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(bdo({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(bdo({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(bdo({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(bdo({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(bdo({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(bdo({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(bdo({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(bdo({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(bdo({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(bdo({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(bdo({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(bdo({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(bdo({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(bdo({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(bdo({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden=true', () => {
        expect(bdo({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden=false', () => {
        expect(bdo({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(bdo({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(bdo({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(bdo({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(bdo({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(bdo({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(bdo({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(bdo({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(bdo({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(bdo({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(bdo({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(bdo({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(bdo({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(bdo({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope=true', () => {
        expect(bdo({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope=false', () => {
        expect(bdo({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(bdo({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(bdo({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(bdo({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(bdo({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(bdo({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(bdo({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style={"color":"#000000"}', () => {
        expect(bdo({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex=1', () => {
        expect(bdo({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(bdo({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(bdo({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(bdo({translate: "no"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(bdo({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(bdo({dir: "rtl"})).toMatchSnapshot();
    });
});