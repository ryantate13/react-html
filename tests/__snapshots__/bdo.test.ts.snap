// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bdo renders arrays of children 1`] = `
<bdo>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</bdo>
`;

exports[`bdo renders with accessKey="a,b,c,d" 1`] = `
<bdo
  accessKey="a,b,c,d"
/>
`;

exports[`bdo renders with autoCapitalize="characters" 1`] = `
<bdo
  autoCapitalize="characters"
/>
`;

exports[`bdo renders with autoCapitalize="none" 1`] = `
<bdo
  autoCapitalize="none"
/>
`;

exports[`bdo renders with autoCapitalize="off" 1`] = `
<bdo
  autoCapitalize="off"
/>
`;

exports[`bdo renders with autoCapitalize="on" 1`] = `
<bdo
  autoCapitalize="on"
/>
`;

exports[`bdo renders with autoCapitalize="sentences" 1`] = `
<bdo
  autoCapitalize="sentences"
/>
`;

exports[`bdo renders with autoCapitalize="words" 1`] = `
<bdo
  autoCapitalize="words"
/>
`;

exports[`bdo renders with children 1`] = `
<bdo>
  TEST
</bdo>
`;

exports[`bdo renders with className="foo bar baz" 1`] = `
<bdo
  className="foo bar baz"
/>
`;

exports[`bdo renders with contentEditable="false" 1`] = `
<bdo
  contentEditable="false"
/>
`;

exports[`bdo renders with contentEditable="true" 1`] = `
<bdo
  contentEditable="true"
/>
`;

exports[`bdo renders with dir="auto" 1`] = `
<bdo
  dir="auto"
/>
`;

exports[`bdo renders with dir="ltr" 1`] = `
<bdo
  dir="ltr"
/>
`;

exports[`bdo renders with dir="ltr" 2`] = `
<bdo
  dir="ltr"
/>
`;

exports[`bdo renders with dir="rtl" 1`] = `
<bdo
  dir="rtl"
/>
`;

exports[`bdo renders with dir="rtl" 2`] = `
<bdo
  dir="rtl"
/>
`;

exports[`bdo renders with draggable="false" 1`] = `
<bdo
  draggable="false"
/>
`;

exports[`bdo renders with draggable="true" 1`] = `
<bdo
  draggable="true"
/>
`;

exports[`bdo renders with empty attributes 1`] = `<bdo />`;

exports[`bdo renders with hidden=false 1`] = `
<bdo
  hidden={false}
/>
`;

exports[`bdo renders with hidden=true 1`] = `
<bdo
  hidden={true}
/>
`;

exports[`bdo renders with id="TEST" 1`] = `
<bdo
  id="TEST"
/>
`;

exports[`bdo renders with inputMode="decimal" 1`] = `
<bdo
  inputMode="decimal"
/>
`;

exports[`bdo renders with inputMode="email" 1`] = `
<bdo
  inputMode="email"
/>
`;

exports[`bdo renders with inputMode="none" 1`] = `
<bdo
  inputMode="none"
/>
`;

exports[`bdo renders with inputMode="numeric" 1`] = `
<bdo
  inputMode="numeric"
/>
`;

exports[`bdo renders with inputMode="search" 1`] = `
<bdo
  inputMode="search"
/>
`;

exports[`bdo renders with inputMode="tel" 1`] = `
<bdo
  inputMode="tel"
/>
`;

exports[`bdo renders with inputMode="text" 1`] = `
<bdo
  inputMode="text"
/>
`;

exports[`bdo renders with inputMode="url" 1`] = `
<bdo
  inputMode="url"
/>
`;

exports[`bdo renders with is="foo" 1`] = `
<bdo
  is="foo"
/>
`;

exports[`bdo renders with itemID="https://google.com/" 1`] = `
<bdo
  itemID="https://google.com/"
/>
`;

exports[`bdo renders with itemProp="https://google.com/" 1`] = `
<bdo
  itemProp="https://google.com/"
/>
`;

exports[`bdo renders with itemRef="foo bar" 1`] = `
<bdo
  itemRef="foo bar"
/>
`;

exports[`bdo renders with itemScope=false 1`] = `
<bdo
  itemScope={false}
/>
`;

exports[`bdo renders with itemScope=true 1`] = `
<bdo
  itemScope={true}
/>
`;

exports[`bdo renders with itemType="https://google.com/" 1`] = `
<bdo
  itemType="https://google.com/"
/>
`;

exports[`bdo renders with lang="" 1`] = `
<bdo
  lang=""
/>
`;

exports[`bdo renders with lang="es-419" 1`] = `
<bdo
  lang="es-419"
/>
`;

exports[`bdo renders with slot="TEST" 1`] = `
<bdo
  slot="TEST"
/>
`;

exports[`bdo renders with spellCheck="false" 1`] = `
<bdo
  spellCheck="false"
/>
`;

exports[`bdo renders with spellCheck="true" 1`] = `
<bdo
  spellCheck="true"
/>
`;

exports[`bdo renders with style={"color":"#000000"} 1`] = `
<bdo
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`bdo renders with tabIndex=1 1`] = `
<bdo
  tabIndex={1}
/>
`;

exports[`bdo renders with title="TEST" 1`] = `
<bdo
  title="TEST"
/>
`;

exports[`bdo renders with translate="no" 1`] = `
<bdo
  translate="no"
/>
`;

exports[`bdo renders with translate="yes" 1`] = `
<bdo
  translate="yes"
/>
`;

exports[`bdo renders without attributes 1`] = `<bdo />`;
